src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/DynamicFlagsIterator.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal abstract class DynamicFlagsIterator
```

Comment:
Why an abstract class? The empty iterator can just be ArrayDynamicFlagsIterator with empty bool[] and 0 position.